package.json

```
@@ -23,7 +23,7 @@
     "reset": "git clean -dfx && yarn",
     "pretest": "git clean -dfX tests/fixtures/untest",
     "posttest": "git clean -dfXq tests/fixtures",
-    "preinstall": "test -z \"${npm_execpath##*yarn*}\""
+    "preinstall": "node -e \"process.exit(process.env.npm_execpath.indexOf('yarn') > -1 ? 0 : 1)\""
```

```suggestion
    "preinstall": "node -e \"process.exit(process.env.npm_execpath.includes('yarn'))\""
```
would work as well :)

---

I opted for well defined exit codes `0` and `1` instead of `true` or `false` 😉 
And I'm not sure if `includes` is supported on all node versions this might be installed with (e.g. 8 or 10?)

---

It is supported (https://node.green/#ES2015-built-in-extensions-String-prototype-methods), but yeah, this is fine too, if you want to merge it as is.

---

`.includes(…)` is supported in Node 8 afaict. And I'd propose to go with `.include` **&** the ternary operator.